ssz/sedes/serializable.py

```diff
@@ -169,21 +170,41 @@ def copy(self, *args, **kwargs):
         ).difference(
             self._meta.field_names[:len(args)]
         )
-        unchanged_kwargs = {
-            key: copy.deepcopy(value)
-            for key, value
-            in self.as_dict().items()
-            if key in missing_overrides
-        }
+        unchanged_kwargs = {}
+        for key, value in self.as_dict().items():
+            if key in missing_overrides:
+                is_immutable_field = (
```

Comment:
couple of things:
- it's probably better to move this to its own function because it's quite a long expression (e.g. `is_immutable_field_value`)
- shouldn't this check be recursive? So that `((1, 2, 3), (4, 5, 6))` counts as immutable as well.
- this seems to count an empty tuple as mutable as well which is incorrect (even though it probably doesn't matter much regarding performance).

- if we have a separate function for this, we can use a dict comprehension:

```Py
unchanged_kwargs = {
    key: value if is_immutable_field_value(value) else copy.deepcopy(value)
    for key, value in self.as_dict().items()
    if key in missing_overrides
}
```

Reply:
Great idea! <3
Applied in https://github.com/ethereum/py-ssz/pull/84/commits/5dbdb2ba667b9521405c2f609d45d041517ddf04